doc/finalization.md

```diff
@@ -0,0 +1,149 @@
+# Finalization
+
+Various node-addon-api methods accept a templated `Finalizer finalizeCallback`
+parameter. This parameter represents a native callback function that runs in
+response to a garbage collection event. A finalizer is considered a _basic_
+finalizer if the callback only utilizes a certain subset of APIs, which may
+provide optimizations, improved execution, or other benefits.
```

Comment:
```suggestion
provide more efficient memory management, optimizations, improved execution, or other benefits.
```

I think we should explicitly point out memory management because that's what caused us to go down this path in the first place.

Reply:
Addressed in [195ec28](https://github.com/nodejs/node-addon-api/pull/1514/commits/195ec28a7b1dcfe571c6df5f8a2e1bb991126ff1).